sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -366,8 +367,9 @@ pi_result cuda_piEventRetain(pi_event event);
 
 /// \endcond
 
-_pi_queue::native_type _pi_queue::get_compute() {
+CUstream _pi_queue::get_next_compute_stream() {
   if (num_compute_streams_ < compute_streams_.size()) {
+    std::lock_guard<std::mutex> guard(compute_stream_mutex_);
     unsigned int idx = num_compute_streams_++;
     if (idx < compute_streams_.size()) {
       PI_CHECK_ERROR(cuStreamCreate(&compute_streams_[idx], flags_));
```

Comment:
Since this incrementation is protected by the mutex, it would be better to only increment it if it is only incremented when there are actually more room in `compute_streams_`, e.g.

```suggestion
    std::lock_guard<std::mutex> guard(compute_stream_mutex_);
    if (num_compute_streams_ < compute_streams_.size()) {
      unsigned int idx = num_compute_streams_++;
      PI_CHECK_ERROR(cuStreamCreate(&compute_streams_[idx], flags_));
```
That way you can trust `num_compute_streams_` as the source of truth when later deciding how many to free.

Also, please add a comment here for why the check is required twice. Should hopefully prevent people from coming by this later and removing it due to it looking redundant.

Reply:
done